src/core/Class.js

```diff
@@ -4,114 +4,112 @@ import * as Util from './Util';
 // @aka L.Class
 
 // @section
-// @uninheritable
 
 // Thanks to John Resig and Dean Edwards for inspiration!
 
-export function Class() {}
-
-Class.extend = function (props) {
-
-	// @function extend(props: Object): Function
-	// [Extends the current class](#class-inheritance) given the properties to be included.
-	// Returns a Javascript function that is a class constructor (to be called with `new`).
-	const NewClass = function (...args) {
+export class Class {
+	_initHooksCalled = false;
 
+	constructor(...args) {
 		Util.setOptions(this);
 
 		// call the constructor
 		if (this.initialize) {
-			this.initialize.apply(this, args);
+			this.initialize(...args);
 		}

```

Comment:
Is it correct to "convert" args twice?

Reply:
Yeahs should be no issue. The `args` are coming in as parameters hence the spread in the constructor to ensure we get an array. Then when calling initialize we want to pass them in the same manner, so they are spread again.

Reply:
Marking this as resolved so we can focus on the open issues.